scripts/o3de/tests/test_repo.py

```diff
@@ -65,6 +65,130 @@
 }
 '''
 
+#This holds the repo 2.0.0 schema, which houses all relevant objects in file
+TEST_O3DE_REPOC_FILENAME = '3b14717bafd5a3bd768d3d0791a44998c3bd0fb2bfa1a7e2ee8bb1a39b04d631.json'
+TEST_O3DE_REPOC_JSON_PAYLOAD = '''
+{
+	"repo_name": "testgem3",
+
+	"origin": "Studios",
+
+    "$schemaVersion":"2.0.0",
+    
+	"repo_uri": "https://downloads.testgem3.com/o3de-repo",
+
+	"summary": "Studios Repository for the testgem3 Gem.",
+
+	"additional_info": "",
+
+	"last_updated": "2023-01-19",
+
+	"gems": [{
+		"gem_name": "testgem3",
+		"display_name": "testgem3 2",
+		"download_api": "HTTP",
+		"license": "Community",
+		"license_url": "https://www.testgem3.com/testgem3-community-license",
+		"origin": "Persistant Studios - testgem3.com",
+		"requirements": "Users will need to download testgem3 Editor from the <a href='https://www.testgem3.com/download/'>testgem3 Web Site</a> to edit/author effects.",
+		"documentation_url": "https://www.testgem3.com/docs/testgem3-v2/plugins/o3de-gem/",
+		"type": "Code",
+		"summary": "The testgem3 Gem provides real-time FX solution for particle effects.",
+		"canonical_tags": [
+			"Gem"
+		],
+		"user_tags": [
+			"Particles",
+			"Simulation",
+			"SDK"
+		],
+		"dependencies": [],
+		"objects": [{
+				"origin_uri": "https://downloads.testgem3.com/o3de-repo/testgem3-2.15/O3DE_testgem3Gem_v2.0.0_Win64_Linux64_Mac64.zip",
+				"version": "2.0.0",
+				"last_updated": "2021-03-09"
+			},
+			{
+				"display_name": "testgem3 2.15.4",
+				"origin_uri": "https://downloads.testgem3.com/o3de-repo/testgem3-2.15/O3DE_testgem3Gem_v2.15.4_Win64_Linux64_Mac64.zip",
+				"version": "2.15.4",
+				"last_updated": "2023-03-09"
+			},
+			{
+				"display_name": "testgem3 3.1.2",
+				"download_api": "GitHub",
+				"origin_uri": "https://github.com/testgem3/O3DEtestgem3Plugin.git",
+				"GitHub": {
+					"download_type": "clone",
+					"commit_hash": "2af04986ed6cdd4518ea044c565de855aac64bfc"
+				},
+				"version": "3.1.2",
+				"last_updated": "2025-02-19"
+			},
+			{
+				"display_name": "testgem3 3.2.0",
+				"download_api": "GitHub",
+				"origin_uri": "https://github.com/testgem3/O3DEtestgem3Plugin",
+				"GitHub": {
+					"download_type": "release",
+					"release_identifier": "3.2.0"
+				},
+				"version": "3.1.2",
+				"last_updated": "2025-02-19"
+			}
```

Comment:
having multiple entries for the same version is interesting.  the problem comes when after a user clones or downloads the object - the gem.json contents that they download don't have this extra data, so it isn't clear if it belongs to a clone, an object downloaded from a remote gem repo, or an object that came from their own internal source control.

Reply:
honestly, I'm thinking this should be an error. There shouldn't be two entries for the same version, otherwise it's just ambiguous

Reply:
either that, or the origin_uri needs to be used to disambiguate the version conflict by considering the source

Reply:
the more I look at the reference format, I think the version specified for testgem 3.2.0 is a typo. For now I'm updating it.